analyzers/buck/buck.go

```diff
@@ -0,0 +1,166 @@
+	analyzer := Analyzer{
+		Module: module,
+		Upload: true,
+		Cmd:    buck.New(module.BuildTarget),
+	}
+	return &analyzer, nil
+}
+
+// Clean is not implemented.
+func (a *Analyzer) Clean() error {
+	return nil
+}
+
+// Build is not implemented.
+func (a *Analyzer) Build() error {
+	return nil
+}
+
+// IsBuilt is not implemented.
+func (a *Analyzer) IsBuilt() (bool, error) {
+	return true, nil
+}
+
+// Analyze analyzes a buck build target and its dependencies.
+func (a *Analyzer) Analyze() (graph.Deps, error) {
+	return a.Cmd.Deps(a.Upload)
+}
+
+// Discover finds a Buck project by first looking for a ".buckconfig" file and then a "BUCK" file.
+// 1. ".buckconfig" file is found and we know that that we are at the root of a Buck cell.
+// 	a. Attempt to find user defined aliases in .buckconfig.
+// 	b. No aliases, run `buck targets //` to find all local targets.
+// 2. "BUCK" file is found.
+// 	a. Run `buck targets <directory>:` to find all local targets.
+func Discover(dir string, opts map[string]interface{}) ([]module.Module, error) {
+	return DiscoverWithCommand(dir, opts, buckCmd)
+}
+
+// DiscoverWithCommand is used to operate Discovery with a custom `buck` command.
+func DiscoverWithCommand(dir string, opts map[string]interface{}, buckCommand func(string, ...string) (string, error)) ([]module.Module, error) {
+	var moduleList []module.Module
+	buckConfig, err := files.Exists(dir, ".buckconfig")
+	if err == nil && buckConfig {
+		file, err := ini.Load(path.Join(dir, ".buckconfig"))
+		if err != nil {
+			return nil, errors.Errorf("Unable to read `.buckconfig`: %s", err)
+		}
+
+		aliases, err := file.GetSection("alias")
+		if err == nil && len(aliases.Keys()) > 0 {
+			for name, target := range aliases.KeysHash() {
+				moduleList = append(moduleList, newModule(name, target, dir))
+			}
+			return moduleList, nil
+		}
+
+		out, err := buckCommand("targets", "//")
+		if err != nil {
+			return nil, err
+		}
+
+		buckTargetList := strings.Split(strings.TrimSpace(out), "\n")
+		for _, target := range buckTargetList {
+			moduleList = append(moduleList, newModule(target, target, dir))
+		}
+		return moduleList, nil
+	}
+
+	buckFile, err := files.Exists(dir, "BUCK")
+	if err == nil && buckFile {
+		wd, err := os.Getwd()
+		if err != nil {
+			return nil, errors.Errorf("Cannot get working directory: %s", err)
+		}
+
+		buckRoot, err := buckCommand("root")
+		if err != nil {
+			return nil, err
+		}
+
+		// Condition the current directory to the format of "parent/child:" from the root directory.
+		buckDirectory := strings.TrimPrefix(wd, strings.TrimSpace(buckRoot)+"/")
+		out, err := buckCommand("targets", buckDirectory+":")
+		if err != nil {
+			return nil, err
+		}
+
+		targets := strings.Split(out, "\n")
+		for _, target := range targets {
+			if len(target) > 0 {
+				moduleList = append(moduleList, newModule(target, target, dir))
+			}
+		}
+	}
+	return moduleList, nil
+}
+
+func buckCmd(cmd string, args ...string) (string, error) {
```

Comment:
I did this to remove the complexity under the hood that is used to get readable results from `buck audit`. I will look into only keeping one `buckCmd` to handle the executable but my original concern is that it will make the code messier, although it introduced the need for `buckRoot` in the `Cmd` struct.